src/Avalonia.Controls.DataGrid/DataGridRow.cs

```diff
@@ -223,10 +223,10 @@ internal DataGrid OwningGrid
         /// <summary>
         /// Index of the row
         /// </summary>
-        internal int Index
+        public int Index
```

Comment:
I think it should be a `DirectProperty` to allow the change to be notified.

Reply:
That need to be confirmed with core team since it might expose more than I expected.

Reply:
This property is probably useless for bindings, if it's not direct. DataGridRow instance is reused across many rows, virtualized.